### PR TITLE
Task/new osgi annotations

### PR DESCRIPTION
**Pull request by EwaFengler**
## Description
Migrate Felix annotations to OSGi in Worker module

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [X] My code follows the [code style](https://github.com/Cognifide/aet/blob/master/CONTRIBUTING.md#coding-conventions) of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

---
I hereby agree to the terms of the AET Contributor License Agreement.

### CODE REVIEW

File: core/worker/src/main/java/com/cognifide/aet/worker/drivers/chrome/ChromeWebDriverFactory.java [hunk 1]
```diff
@@ -52,37 +37,28 @@
...
-      description = NAME_DESC,
-      value = DEFAULT_BROWSER_NAME)
-  private String name;
-
-  @Property(name = SELENIUM_GRID_URL,
-      label = SELENIUM_GRID_URL_LABEL,
-      description = "Url to selenium grid hub. When null local Chrome driver will be used. Local Chrome driver does not work on Linux",
-      value = DEFAULT_SELENIUM_GRID_URL)
-  private String seleniumGridUrl;
+  private ChromeWebDriverFactoryConf chromeWebDriverFactoryConf;
```
Comment (Skejven) [inline core/worker/src/main/java/com/cognifide/aet/worker/drivers/chrome/ChromeWebDriverFactory.java]: In PRs for other modules I've seen convention of naming those properties `config`. Could we please keep it also here?

File: core/worker/src/main/java/com/cognifide/aet/worker/drivers/WebDriverHelper.java [hunk 2]
```diff
@@ -27,22 +27,18 @@
...
-
-  public static final String PATH_LABEL = "Path";
+  public static final String PATH_LABEL = "Custom path to Firefox binary";
 
-  public static final String PATH_DESC = "Custom path to driver binary";
-
-  public static final String SELENIUM_GRID_URL = "seleniumGridUrl";
+  public static final String LOG_FILE_PATH_LABEL = "Path to firefox error log";
 
   public static final String SELENIUM_GRID_URL_LABEL = "Selenium grid URL";
```
Comment (Skejven) [inline core/worker/src/main/java/com/cognifide/aet/worker/drivers/WebDriverHelper.java]: It look that those constants are only used in `WebDriverProviderConf`. I believe they should be defined here.
This `WebDriverHelper` look like generic class - so there should be no constants that have `Firefox` or `Chrome` words in it.

File: core/worker/src/main/java/com/cognifide/aet/worker/drivers/firefox/local/FirefoxWebDriverFactory.java [hunk 3]
```diff
@@ -1,81 +1,58 @@
 /**
  * AET
- *
+ * <p>
```
Comment (Skejven) [inline core/worker/src/main/java/com/cognifide/aet/worker/drivers/firefox/local/FirefoxWebDriverFactory.java]: Please remove this formatting changes form license header.

File: core/worker/src/main/java/com/cognifide/aet/worker/drivers/WebDriverProvider.java [hunk 4]
```diff
@@ -23,45 +23,46 @@
...
 
   @Activate
-  void activate(Map<String, String> properties) {
-    defaultWebDriverName = properties.get(DEFAULT_WEB_DRIVER_NAME);
+  void activate(WebDriverProviderConf webDriverProviderConf) {
+    this.webDriverProviderConf = webDriverProviderConf;
   }
 
-  public WebCommunicationWrapper createWebDriverWithProxy(String preferredWebDriver, String proxyName)
+  public WebCommunicationWrapper createWebDriverWithProxy(String preferredWebDriver,
```
Comment (Skejven) [inline core/worker/src/main/java/com/cognifide/aet/worker/drivers/WebDriverProvider.java]: For the future could you please don't change formatting on the lines that didn't change during the refactoring?
Please create additional PR to fix only formatting.

File: core/worker/src/main/java/com/cognifide/aet/worker/drivers/WebDriverProvider.java [hunk 5]
```diff
@@ -23,42 +23,46 @@
...
+@Designate(ocd = WebDriverProviderConfig.class)
 public class WebDriverProvider {
 
   private static final Logger LOG = LoggerFactory.getLogger(WebDriverProvider.class);
 
-  private static final String DEFAULT_WEB_DRIVER_NAME = "defaultWebDriverName";
-
-  @Property(name = DEFAULT_WEB_DRIVER_NAME, label = "Default Web Driver name", value = "ff")
-  private String defaultWebDriverName;
+  private WebDriverProviderConfig webDriverProviderConfig;
```
Comment (Skejven) [inline core/worker/src/main/java/com/cognifide/aet/worker/drivers/WebDriverProvider.java]: In PRs for other modules I've seen convention of naming those properties `config`. Could we please keep it also here and other classes in this PR?

File: core/worker/pom.xml [hunk 6]
```diff
@@ -71,6 +71,18 @@
...
+      <artifactId>org.osgi.service.component.annotations</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.osgi</groupId>
+      <artifactId>org.osgi.annotation</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.osgi</groupId>
+      <artifactId>org.osgi.service.metatype.annotations</artifactId>
+    </dependency>
```
Comment (Skejven) [inline core/worker/pom.xml]: You can also remove:
```xml
    <dependency>
      <groupId>org.apache.sling</groupId>
      <artifactId>org.apache.sling.commons.osgi</artifactId>
    </dependency>
```

File: core/worker/src/main/java/com/cognifide/aet/worker/drivers/WebDriverProvider.java (hunk 5 above)
Comment (EwaFengler) [reply core/worker/src/main/java/com/cognifide/aet/worker/drivers/WebDriverProvider.java]: You mean changing this to 
`private WebDriverProviderConfig config`?

File: core/worker/pom.xml (hunk 6 above)
Comment (EwaFengler) [reply core/worker/pom.xml]: No, I can't - WebDriverHelper uses PropertiesUtil from this package.

Comment (PiteroS8) [reply core/worker/pom.xml]: Maybe -> `return StringUtils.defaultString(properties.get(name), defaultValue);` at https://github.com/Cognifide/aet/blob/a024722d58c9704720199560a19ae4897058f555/core/worker/src/main/java/com/cognifide/aet/worker/drivers/WebDriverHelper.java#L39

File: core/worker/src/main/java/com/cognifide/aet/worker/drivers/WebDriverProvider.java (hunk 5 above)
Comment (Skejven) [reply core/worker/src/main/java/com/cognifide/aet/worker/drivers/WebDriverProvider.java]: Yes, and all other `XYZConfig` fields in other services.

File: core/worker/pom.xml (hunk 6 above)
Comment (Skejven) [reply core/worker/pom.xml]: Is this method used (`WebDriverHelper.getProp`)? Some of this method references were removed from `ChromeWebDriverFactory.activate`.

**Pull request merged**